sky/skylet/providers/azure/config.py

```diff
@@ -126,9 +130,27 @@ def _configure_resource_group(config):
         .properties.outputs
     )
 
+    # We should wait for the NSG to be created before opening any ports
+    # to avoid overriding the newly-added NSG rules.
+    nsg_id = outputs["nsg"]["value"]
+    nsg_name = nsg_id.split("/")[-1]
+    network_client = NetworkManagementClient(credentials, subscription_id)
+    backoff = common_utils.Backoff()
```

Comment:
Backoff for this kind of retry seems not needed and may increase the waiting time. Should we set `max_backoff_factor=1` to avoid increasing the sleep? (backoff should only be useful for the case when the request is failed due to network issues or rate limit)

Reply:
Done. Thanks!